### PR TITLE
Add a --verbose-build option to the deploy script.

### PR DESCRIPTION
**Pull request by gendx**
This pull request adds a `--verbose-build` option to the deploy script, to print detailed output about the various building steps.

- [x] Tests pass
- [x] Appropriate changes to README are included in PR

### CODE REVIEW

File: deploy.py [hunk 1]
```diff
@@ -711,6 +735,13 @@ def main(args):
       help=("Only compiles and flash the application/example. "
             "Otherwise TockOS will also be bundled and flashed."),
   )
+  main_parser.add_argument(
+      "--verbose-build",
+      action="store_true",
+      default=False,
+      dest="verbose_build",
+      help=("Build everything in verbose mode."),
```
Comment (jmichelp) [inline deploy.py:743]: The parentheses are not needed

File: deploy.py [hunk 2]
```diff
@@ -300,10 +308,18 @@ def update_rustc_if_needed(self):
             target_toolchain[1] in current_version):
       info("Updating rust toolchain to {}".format("-".join(target_toolchain)))
       # Need to update
-      self.checked_command_output(
-          ["rustup", "install", target_toolchain_fullstring])
-    self.checked_command_output(
-        ["rustup", "target", "add", SUPPORTED_BOARDS[self.args.board].arch])
+      rustup_install = ["rustup"]
+      if self.args.verbose_build:
+        rustup_install.extend(["--verbose"])
```
[the remark this answers is not in the document]
Comment (gendx) [reply deploy.py:313]: Nope: `fatal: Failed to execute rustup: Command '['rustup', '-', '-', 'v', 'e', 'r', 'b', 'o', 's', 'e', 'install', 'nightly-2020-02-03']' returned non-zero exit status 1.`

File: deploy.py [hunk 3]
```diff
@@ -300,10 +308,18 @@ def update_rustc_if_needed(self):
...
+      rustup_install.extend(["install", target_toolchain_fullstring])
+      self.checked_command(rustup_install)
+
+    rustup_target = ["rustup"]
+    if self.args.verbose_build:
+      rustup_target.extend(["--verbose"])
```
[the remark this answers is not in the document]
Comment (gendx) [reply deploy.py:319]: Nope: `fatal: Failed to execute rustup: Command '['rustup', '-', '-', 'v', 'e', 'r', 'b', 'o', 's', 'e', 'install', 'nightly-2020-02-03']' returned non-zero exit status 1.`

File: deploy.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gendx) [reply deploy.py]: My bad, I forgot to change `extend` -> `append`.

**Pull request merged**